app/jobs/session_trim_job.rb

```diff
@@ -0,0 +1,17 @@
+# frozen_string_literal: true
+
+class SchoolDataImporterJob < CronJob
+  self.cron_expression = "0 0 1 * *"
+
+  queue_as :session_trim
+
+  def perform
+    Rails.logger.info "Trimming session store..."
+
+    cutoff_period = ENV.fetch("SESSION_DAYS_TRIM_THRESHOLD", 30).to_i.days.ago
+
+    ActiveRecord::SessionStore::Session
+      .where("updated_at < ?", cutoff_period)
+      .delete_all
```

Comment:
Is this the best logic? Do we have access to sessions here? I don't know how Rails handles them (never really looked into that), but I think we could do something like 'fetch all session store data that have id pointing at an expired or deleted session, and drop them'. Which would mean we would not need to worry about timing - we would simply defer that logic to session logic.

That said, it is a bit surprising to me that the gem doesn't automatically delete session store when session expires / is deleted. Thoughts?

Reply:
So the gem actually defines a rake task `db:session:trim` for cleaning up the session, this logic was copied from that task (Wasn't sure if we should load the task into the web server logic). Alternative was to invoke rake via a system call or, ideally add rake task call to the cron schedule (do we need rufus-scheduler for that?)

So the way sessions works is that we still keep session id within a cookie (with expiry date), but session record itself has no expiration date, which is why they suggest running that rake regularly to prevent uncontrolled session table growth. I agree, we could do it better - we can replace the default Session record definition with our custom one, but I think for now the default is good enough.